Makefile

```diff
@@ -3,6 +3,10 @@ DOCKER_TAG ?= $(shell git rev-parse --short HEAD)
 DOCKER_IMAGE := $(DOCKERHUB_REPO):$(DOCKER_TAG)
 DOCKER_IMAGE_LATEST := $(DOCKERHUB_REPO):latest
 
+# NOTE: The `DOCKER_IMAGE_BRANCH` tag is the git tag for the commit if it exists, else the branch on which the commit exists.
+# NOTE: Special characters in `DOCKER_IMAGE_BRANCH` are replaced with dashes.
+DOCKER_IMAGE_BRANCH := $(DOCKERHUB_REPO):$(shell git describe --exact-match --tags 2> /dev/null || git symbolic-ref --short HEAD | sed 's/[^[:alnum:]\.\_\-]/-/g')
```

Comment:
This is clean and produces less tags. I should apply this knowledge (https://github.com/TACC/TACC-Docs/pull/25#issuecomment-2123602887).

Reply:
Yes! That also helped fixed bug. Thanks original author 🙂.